superset/db_engine_specs/base.py

```diff
@@ -1724,6 +1726,18 @@ def get_public_information(cls) -> Dict[str, Any]:
             "disable_ssh_tunneling": cls.disable_ssh_tunneling,
         }
 
+    @classmethod
+    def validate_database_uri(cls, sqlalchemy_uri: URL) -> None:
+        """
+        Validates a database SQLAlchemy URI per engine spec.
+        Use this to implement a final validation for unwanted connection configuration
+
+        :param sqlalchemy_uri:
+        """
+        for query_param in sqlalchemy_uri.query.keys():
+            if query_param in cls.disallow_uri_query_params:
+                raise ValueError("Disallowed query parameter")
```

Comment:
I agree with @Antonio-RiveroMartnez , it would be nice to list the query params that are disallowed. To do it all in one go one could use `set1.intersection(set2)` to get all overlapping disallowed params to avoid iterating over the list.

Reply:
nice!